### PR TITLE
OLS-888: Install operator from bundle

### PR DESCRIPTION
**Pull request by onmete**
## Description

Install operator from bundle instead catalog.

## Type of change

- [x] Refactor
- [x] New feature

### CODE REVIEW

File: tests/e2e/test_api.py [hunk 1]
```diff
@@ -85,21 +86,29 @@ def install_ols() -> tuple[str, str, str]:
...
+    try:
+        subprocess.run(  # noqa: S603
+            [  # noqa: S607
+                "operator-sdk",
+                "run",
+                "bundle",
+                "--timeout=10m",
+                "-n",
+                "openshift-lightspeed",
+                "quay.io/openshift-lightspeed/lightspeed-operator-bundle:latest",
```
Comment (raptorsun) [inline tests/e2e/test_api.py:100]: try add a `--verbose` to see more information for debugging

File: tests/e2e/test_api.py [hunk 2]
```diff
@@ -85,21 +86,29 @@ def install_ols() -> tuple[str, str, str]:
...
-    print("Created catalog+subscription")
+    # install the operator from bundle
+    print("Installing OLS operator from bundle")
+    try:
+        subprocess.run(  # noqa: S603
+            [  # noqa: S607
+                "operator-sdk",
+                "run",
+                "bundle",
+                "--timeout=10m",
```
Comment (raptorsun) [inline tests/e2e/test_api.py:97]: we can try increase it to 20 minutes to see if that helps, maybe the CI cluster is slower than normal

File: tests/config/operator_install/imagedigestmirrorset.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# The images are pulled from our redhat-user-workloads registry or staging registry while the production image is not yet available.
+kind: ImageDigestMirrorSet
+apiVersion: config.openshift.io/v1
+metadata:
+  name: openshift-lightspeed-prod-to-ci
+spec:
+  imageDigestMirrors:
+    - source: registry.redhat.io/openshift-lightspeed-beta/lightspeed-rhel9-operator
+      mirrors:
+        - registry.staging.redhat.io/openshift-lightspeed-beta/lightspeed-rhel9-operator
```
Comment (xrajesh) [inline tests/config/operator_install/imagedigestmirrorset.yaml:10]: Remove the above line - same goes to all the ones `- registry.staging.redhat.io` below.

File: tests/config/operator_install/imagedigestmirrorset.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        - quay.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-operator
+    - source: registry.redhat.io/openshift-lightspeed-beta/lightspeed-operator-bundle
+      mirrors:
+        - registry.staging.redhat.io/openshift-lightspeed-beta/lightspeed-operator-bundle
```
Comment (xrajesh) [inline tests/config/operator_install/imagedigestmirrorset.yaml:14]: Remove the above line

File: tests/config/operator_install/imagedigestmirrorset.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        - quay.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/bundle
+    - source: registry.redhat.io/openshift-lightspeed-beta/lightspeed-service-api-rhel9
+      mirrors:
+        - registry.staging.redhat.io/openshift-lightspeed-beta/lightspeed-service-api-rhel9
```
Comment (xrajesh) [inline tests/config/operator_install/imagedigestmirrorset.yaml:18]: Remove the above line

File: tests/config/operator_install/imagedigestmirrorset.yaml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        - quay.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-service
+    - source: registry.redhat.io/openshift-lightspeed-beta/lightspeed-console-plugin-rhel9
+      mirrors:
+        - registry.staging.redhat.io/openshift-lightspeed-beta/lightspeed-console-plugin-rhel9
```
Comment (xrajesh) [inline tests/config/operator_install/imagedigestmirrorset.yaml:22]: Remove the above line

**Comment by xrajesh** [comment]
/override "Red Hat Konflux / ols-enterprise-contract / lightspeed-service"

**Pull request merged**